File: Tutorial_Sample_Code/DataConnectivity/IDataReader/VB/VB_Win_Data_IDataReader/Form1.vb
Imports CrystalDecisions.CrystalReports.Engine
Imports CrystalDecisions.Shared

Public Class Form1
    Private customersViaIdrReport As ReportDocument

    Private Sub ConfigureCrystalReports()
        customersViaIdrReport = New ReportDocument()
        Dim reportPath As String = Application.StartupPath & "\" & "CustomersViaIDR.rpt"
        customersViaIdrReport.Load(reportPath)
        myCrystalReportViewer.ReportSource = customersViaIdrReport
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ConfigureCrystalReports()
    End Sub
End Class

File: Tutorial_Sample_Code/DataConnectivity/IDataReader/CS/CS_Win_Data_IDataReader/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Shared;

namespace CS_Base
{
    public partial class Form1 : Form
    {
        private ReportDocument customersViaIdrReport;
        public Form1()
        {
            InitializeComponent();
        }

        private void ConfigureCrystalReports()
        {
            customersViaIdrReport = new ReportDocument();
            string reportPath = Application.StartupPath + "\\" + "CustomersViaIDR.rpt";
            customersViaIdrReport.Load(reportPath);
            crystalReportViewer.ReportSource = customersViaIdrReport;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            ConfigureCrystalReports();
        }

        private void crystalReportViewer_Load(object sender, EventArgs e)
        {

        }
    }
}